sw/device/lib/dif/dif_uart.h

```diff
@@ -365,6 +382,41 @@ dif_uart_result_t dif_uart_rx_bytes_available(const dif_uart_t *uart,
 DIF_WARN_UNUSED_RESULT
 dif_uart_result_t dif_uart_tx_bytes_available(const dif_uart_t *uart,
                                               size_t *num_bytes);
+/**
+ * UART TX reset RX/TX FIFO.
+ *
+ * Reset both FIFOs, or the requested one.
+ *
+ * @param uart UART state data.
+ * @param reset FIFO to reset (RX or TX).
+ * @return `dif_uart_result_t`.
+ */
+dif_uart_result_t dif_uart_fifo_reset(const dif_uart_t *uart,
```

Comment:
Can you please add some details about what happens to in-progress TX? I assume that it stops transmitting, but it would be useful to document that here.

Reply:
Sure, good point, thanks.

Reply:
I am not entirely sure what is the behaviour of the RX/TX reset. From what I gathered from RTL - RXRST and TXRST only affects the FIFO, nothing else. So it seems to me that even if there is a TX/RX in progress, RX/TXRST won't clobber the shift register, and If the transmission is ongoing you would eventually get a byte in the FIFO.

There is a high chance that I have misunderstood the code in the relevant `*.sv` files, so I will submit an issue to clarify it with UART D2 reviewers.

Meanwhile I will merge this change, as it probably falls under S2 checklist anyway.